### PR TITLE
feather-wallet: init at 2.1.0

### PR DESCRIPTION
**Pull request by neverupdate**
###### Description of changes

A lightweight wallet for the Monero cryptocurrency
https://github.com/feather-wallet/feather

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/blockchains/feather-wallet/default.nix [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    qtbase qtwebsockets qtsvg
+    miniupnpc unbound readline
+    randomx libgcrypt libgpg-error
+    boost libunwind libsodium pcsclite
+    zeromq hidapi rapidjson qrencode zbar expat
+    libusb1 protobuf libzip git
+  ];
+
+  patches = [
+    ./config_fix.patch
```
Comment (onny) [inline pkgs/applications/blockchains/feather-wallet/default.nix:39]: A small comment here on why this patch is needed would be helpful :)

File: pkgs/applications/blockchains/feather-wallet/install_fix.patch [hunk 2]
```diff
@@ -0,0 +1,11 @@
+diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
+index 481874d..6dbe2b2 100644
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -302,5 +302,5 @@ if(APPLE)
+ endif()
+ 
+ install(TARGETS feather
+-        DESTINATION ${CMAKE_INSTALL_PREFIX}
++        DESTINATION ${CMAKE_INSTALL_PREFIX}/bin
```
Comment (onny) [inline pkgs/applications/blockchains/feather-wallet/install_fix.patch:10]: Is it possible to set INSTALL_PREFIX via cmakeFlags?

File: pkgs/applications/blockchains/feather-wallet/default.nix [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+, randomx, zeromq, libgcrypt, libgpg-error
+, hidapi, rapidjson, qrencode, zbar, expat
+, libzip, libusb1, protobuf, python3
+}:
+
+stdenv.mkDerivation rec {
+  pname = "feather-wallet";
+  version = "2.1.0";
+
+  src = fetchgit {
```
Comment (onny) [inline pkgs/applications/blockchains/feather-wallet/default.nix:16]: Consider using function `fetchFromGitHub`

File: pkgs/applications/blockchains/feather-wallet/default.nix [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    url = "https://github.com/feather-wallet/feather.git";
+    rev    = "${version}";
```
Comment (onny) [inline pkgs/applications/blockchains/feather-wallet/default.nix:18]: You can do `rev = version;`

File: pkgs/applications/blockchains/feather-wallet/config_fix.patch [hunk 5]
```diff
@@ -0,0 +1,12 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt                                                                                                 
+index ce8c9fb..93b85a4 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -152,6 +152,7 @@ if(TOR_DIR)
+     message(STATUS "Embedding Tor binaries at ${TOR_DIR}")
+ else()
+     message(STATUS "Skipping Tor inclusion because -DTOR_DIR=Off")
++    configure_file("cmake/config-feather.h.cmake" "${CMAKE_CURRENT_SOURCE_DIR}/src/config-feather.h")
```
Comment (onny) [inline pkgs/applications/blockchains/feather-wallet/config_fix.patch:9]: Is this an issue we could also report upstream?

**Review by onny — COMMENTED** [review]
Thank you for your first contribution :) Looks already good, only some minor notes.

File: pkgs/applications/blockchains/feather-wallet/install_fix.patch (hunk 2 above)
Comment (neverupdate) [reply pkgs/applications/blockchains/feather-wallet/install_fix.patch:10]: I think the `CMAKE_INSTALL_PREFIX` is fine, the problem is that the executable was being installed at the root, not in the `${CMAKE_INSTALL_PREFIX}/bin` directory. I should probably report this to upstream.

File: pkgs/applications/blockchains/feather-wallet/default.nix (hunk 3 above)
Comment (neverupdate) [reply pkgs/applications/blockchains/feather-wallet/default.nix:16]: The build process needs the `.git` directory to get version information, so I had to use `fetchgit` with the `leaveDotGit = true` option. I don't know if `fetchFromGitHub` has this option.

File: pkgs/applications/blockchains/feather-wallet/config_fix.patch (hunk 5 above)
Comment (neverupdate) [reply pkgs/applications/blockchains/feather-wallet/config_fix.patch:9]: Yes, someone already reported it here: https://github.com/feather-wallet/feather/issues/33

File: pkgs/applications/blockchains/feather-wallet/default.nix (hunk 4 above)
Comment (neverupdate) [reply pkgs/applications/blockchains/feather-wallet/default.nix]: Fixed!

File: pkgs/applications/blockchains/feather-wallet/default.nix (hunk 1 above)
Comment (neverupdate) [reply pkgs/applications/blockchains/feather-wallet/default.nix:39]: Comments added.

**Comment by kurnevsky** [comment]
Got the following error when tried to build:
```
error: hash mismatch in fixed-output derivation '/nix/store/8azf9ljrd559f3jam8gpyd38fc8syp2z-feather.drv':
         specified: sha256-gPNDd4hRwRCAQcvF3B+LG3tYFBkvFl5uIuF50CLcPm8=
            got:    sha256-chhuqWTcglsuDkjVsxdM72oty1gHZPisMvyPdbJPBnM=
```

File: pkgs/applications/blockchains/feather-wallet/default.nix (hunk 3 above)
Comment (onny) [reply pkgs/applications/blockchains/feather-wallet/default.nix:16]: `leaveDotGit` should also work for `fetchFromGitHub` https://github.com/NixOS/nixpkgs/blob/350fd0044447ae8712392c6b212a18bdf2433e71/pkgs/applications/virtualization/ignite/default.nix#L21 
Would be cool to use this!

**Comment by onny** [comment]
Thanks for fixing most of the issues :) Please put the changes to the maintainers list into a separate commit with the title `maintainers: add neverupdate`

**Comment by surfaceflinger** [comment]
Not sure what's happening with this PR but hash is changing very often. Maybe because of git submodules, but these are pinned to revs by git.

**Comment by ekimber** [comment]
I updated this for 2.3.0, the config_patch is no longer needed as that issue is fixed. However I cannot get anywhere with it, also because the hashes change all the time due to the submodules. I don't see where they are pinned? I'm just going to use appimage-run for this software

**Comment by kurnevsky** [comment]
> I don't see where they are pinned?

It should be pinned by git, somewhere inside `.git` I guess.

**Comment by abathur** [comment]
The hash problems mentioned here might be a bug. Check out:
- issue: #240797
- candidate fix: #249733

This may be tractable once those land (and I guess this case may also help validate and thus speed up the fix)?

**Comment by surfaceflinger** [comment]
^looks like these issues are now resolved, @neverupdate are you still here or should I try making a new PR and possibly maintain it?

**Comment by pbsds** [comment]
Package was added in #265694

**Pull request closed without merging**